src/manager.rs

```diff
@@ -181,13 +184,15 @@ impl AuthenticatorTransport for U2FManager {
             status,
             callback,
         };
-        self.tx.send(action).map_err(|_| crate::Error::Unknown)
+        self.tx
+            .send(action)
+            .map_err(|e| AuthenticatorError::from(e))
     }
 
-    fn cancel(&mut self) -> Result<(), crate::Error> {
+    fn cancel(&mut self) -> crate::Result<()> {
         self.tx
             .send(QueueAction::Cancel)
-            .map_err(|_| crate::Error::Unknown)
+            .map_err(|e| AuthenticatorError::from(e))
```

Comment:
Isn't this `from` call what `?` does already? (see https://doc.rust-lang.org/std/convert/trait.From.html#examples). Something like `Ok(self.tx.send(...)?)` is terser, unless you think the explicit call to `from` is worth keeping.

(Though I think ideally this would look like `self.tx.send(...).into()` but it seems the necessary impl to make this work isn't in std)

Reply:
Aha, I didn't even _look_ at this closely, thank you!